src/nvme/ioctl.c

```diff
@@ -503,7 +504,7 @@ static int read_ana_chunk(int fd, enum nvme_log_ana_lsp lsp, bool rae,
 	}
 
 	while (*read < to_read) {
-		__u32 len = min(log_end - *read, NVME_LOG_PAGE_PDU_SIZE);
+		__u32 len = min((int)(log_end - *read), NVME_LOG_PAGE_PDU_SIZE);
```

Comment:
I think it would be better to use `min_t` here:

    `min_t(int, log_end - *read, NVME_LOG_PAGE_PDU_SIZE);`

Reply:
thanks, make sense.

Reply:
Sorry, thinko on my side. The final type is `__u32`, so we should also cast to `__u32` and not `int`.

Reply:
the result of subtracting two u8* pointers is a signed integer,  use `int` also makes sense